scripts/test-upgrade.sh

```diff
@@ -18,34 +18,65 @@
 
 set -eux -o pipefail
 
-BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
-OA_DIR="$BASE_DIR/openstack-ansible"
-RPCD_DIR="$BASE_DIR/rpcd"
+export BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
+export OA_DIR="$BASE_DIR/openstack-ansible"
+export RPCD_DIR="$BASE_DIR/rpcd"
+
+source ${BASE_DIR}/scripts/functions.sh
+
+# Create a backup of /etc/openstack_deploy, if one doesn't already exist
+# NOTE: This gets done by openstack-ansible's scripts/run-upgrade.sh script,
+#       but if we wait until then it'll be too late.
+if [[ ! -d "/etc/openstack_deploy.MITAKA" ]]; then
```

Comment:
Looks like this is still referencing the old `MITAKA`.  Side note, I created https://review.openstack.org/#/c/369650/ so that upstream OSA upgrade backs stuff up to /etc/openstack_deploy.LIBERTY, rather than /etc/openstack_deploy.MITAKA.  In liberty of OSA it backed up to /etc/openstack_deploy.KILO.

With that said, I almost feel like this script should back up somewhere else, ie. /etc/openstack_deploy.LIBERTY_RPCO or whatever.  The reason being that when OSA upgrade commences and backs up these files it's going to overwrite the original backups we had in /etc/openstack_deploy.LIBERTY.  It would be good to have a pristine backup of /etc/openstack_deploy which we know OSA won't touch that support can resort to if needed.  Thoughts?